google/cloud/storage/blob.py

```diff
@@ -3407,6 +3412,115 @@ def update_storage_class(
                 retry=retry,
             )
 
+    def open(
+        self,
+        mode="r",
+        chunk_size=None,
+        encoding=None,
+        errors=None,
+        newline=None,
+        **kwargs
+    ):
+        r"""Create a file handler for file-like I/O to or from this blob.
+
+        This method can be used as a context manager, just like Python's
+        built-in 'open()' function.
+
+        :type mode: str
```

Comment:
I think `mode` should be a required parameter to not confuse devs.

Reply:
This follows the custom of open() defaulting to read used in the Python standard library. However, it's not that important that we follow that custom in this case. Given that, do you still think we should change it to required?

Reply:
Oh, I didn't realize that open() in Python defaulted to reads. I feel like we should be opinionated because it turns into one of two types which has different implications. I'm open to leaving as-is, but would lean towards being required given the underlying complexities.